From b9809d478d792776fd6bb79b94732e11f3ed3d59 Mon Sep 17 00:00:00 2001
From: Jonh Wendell <jonh.wendell@redhat.com>
Date: Tue, 23 Feb 2021 14:41:44 -0500
Subject: [mec] MAISTRA-2140: Don't use pre-compiled WASM extensions (#272)

* MAISTRA-1475: Don't use pre-compiled WASM extensions

We are not going to use them for 2.1, because they are built
on x86 architecture, hence they will not work on other arches.

So, let's use normal, arch-independent wasm files.

* make gen
---
 .../templates/telemetryv2_1.10.yaml           | 24 +++++++++----------
 .../templates/telemetryv2_1.11.yaml           | 24 +++++++++----------
 .../templates/telemetryv2_1.12.yaml           | 24 +++++++++----------
 .../templates/telemetryv2_1.10.yaml           | 24 +++++++++----------
 .../templates/telemetryv2_1.11.yaml           | 24 +++++++++----------
 .../templates/telemetryv2_1.12.yaml           | 24 +++++++++----------
 6 files changed, 72 insertions(+), 72 deletions(-)

diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.10.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.10.yaml
index 65f0eddf83..6da75dd66f 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.10.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.10.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
index fba3a5ec22..8c0b380e96 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.11.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
diff --git a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
index aeb987f23e..086a5246d9 100644
--- a/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
+++ b/manifests/charts/istio-control/istio-discovery/templates/telemetryv2_1.12.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.10.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.10.yaml
index 65f0eddf83..6da75dd66f 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.10.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.10.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
index fba3a5ec22..8c0b380e96 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.11.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
diff --git a/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml b/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
index aeb987f23e..086a5246d9 100644
--- a/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
+++ b/manifests/charts/istiod-remote/templates/telemetryv2_1.12.yaml
@@ -51,10 +51,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -106,10 +106,10 @@ spec:
                   vm_id: stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -153,10 +153,10 @@ spec:
                   vm_id: stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -220,10 +220,10 @@ spec:
                   vm_id: tcp_stats_inbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -264,10 +264,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
@@ -308,10 +308,10 @@ spec:
                   vm_id: tcp_stats_outbound
                   {{- if .Values.telemetry.v2.prometheus.wasmEnabled }}
                   runtime: envoy.wasm.runtime.v8
-                  allow_precompiled: true
+                  allow_precompiled: false
                   code:
                     local:
-                      filename: /etc/istio/extensions/stats-filter.compiled.wasm
+                      filename: /etc/istio/extensions/stats-filter.wasm
                   {{- else }}
                   runtime: envoy.wasm.runtime.null
                   code:
-- 
2.31.1

